tools/functional-tester/etcd-agent/agent.go

```diff
@@ -102,9 +102,11 @@ func (a *Agent) restart() error {
 }
 
 func (a *Agent) cleanup() error {
-	err := a.stop()
-	if err != nil {
-		return err
+	if a.state == stateStarted {
```

Comment:
The cleanup may happen when the etcd process is stopped, and this can lead to error 'stopping an stopped process'
So I think it should only stop etcd if etcd is in stateStarted.